### DISCUSSION TITLE
How to make a query parameter optional?

### DISCUSSION THREAD

**Discussion by clv-wchaspuri** [2sic/2sxc#3393] [Q&A]
Hello, 

I have a query that contains a value filter using the query parameter searchText.  It works great filtering item, but it doesn't return anything when I don't include the query parameter in the URL

For example, if I have 
/News?searchText=summer   - returns all item with a title container the word summer
/News                                     - returns empty

![image](https://github.com/2sic/2sxc/assets/75698335/94321d44-0443-4be3-af61-3875095c24e8)

Any input and help would be greatly appreciated, thank you!

**Comment by iJungleboy** [comment]
You have a few options. 

1. Include a fallback value to the token, eg `[Params:search||default-title-to-search]`
2. use the fallback stream which is the one used when nothing is found

**Reply by clv-wchaspuri** [reply]
Thanks! that kind of worked, but now, is there a way to make it return nothing when no item matches the search query?

So now what I have is

News?searchText=summer - returns news with the title 'summer'.
News - returns the full list.
News?searchText=asdf - also returns the full list.

**Reply by iJungleboy** [reply]
good point

No, ATM the fallback is for when it matches nothing, which is kind of what happens here. 

The other option doesn't really exist ATM, but I do see that it's clearly a shortcoming of the concept. 

I suggest you create an issue for this.
